src/TuxAnimations.js

```diff
@@ -45,75 +46,127 @@ var makeAnimation = function (transitions, customClassName) {
-            if (typeof delay === "number") {
-              delay = delay + "ms";
+            //Delay prop work--------------------------------------------
+            var delay = this.props.delay;
+            //Use default if no delay prop defined
+            if (delay) {
+              //Case to accept if delay prop is a number
+              if (typeof delay === "number") {
+                delay = delay + "ms";
+              }
+              transitions[action]['transition-delay'] =  delay;
             }
-            transitions[action]['transition-delay'] =  delay;
-          }
 
-          //Custom properties work-------------------------------------
-          var custom = this.props.custom;
-          if (custom) {
-            for (var cssElement in custom) {
-              transitions['enterActive'][cssElement] =  custom[cssElement];
-              transitions['leave'][cssElement] =  custom[cssElement];
+            //Custom properties work-------------------------------------
+            var custom = this.props.custom;
+            if (custom) {
+              for (var cssElement in custom) {
+                transitions['enter-active'][cssElement] =  custom[cssElement];
+                transitions['leave'][cssElement] =  custom[cssElement];
+              }
             }
-          }
-        }
+          }//End for css in transitions[action]
+        }//End if action === className
       }//End for loop for transitions
+    },
+
+    componentWillEnter: function (callback) {
+      this.setAnimationDomNode('enter', callback);
+    },
 
+    componentWillLeave: function (callback) {
+      this.setAnimationDomNode('leave', callback);
+    },
+
+    render: function () {
       //Return new React.Dom element
       return (
         React.DOM.div(
           {
             className: className
           },
-           this.props.children
+          this.props.animate
         )
       );
     }
    });
 };
 
+// makeAnimationGroup FUNCTION: wraps custom animation component in a ReactTransitionGroup by changin
+  // @param Node OBJECT: animation component based on custom properties
+var makeAnimationGroup = function (Node) {
+  //React.createClass FUNCTION: function to create animation component
+    //@param OBJECT: component setState, props, and render
+  return React.createClass({
+    // Set toAnimate initial state as an array that wraps custom Node component
+    // *** toAnimate is an array because ReactTransitionGroup only accepts multiple elements if wrapped in a single array
+    getInitialState: function () {
+      return {
+        toAnimate: []
+      }
+    },
+    // componenetWillRecieveProps FUNCTION: standard React function to update properties via state
+      // @param newProps OBJECT: new properties that will update the toAnimate state.
+    componentWillReceiveProps: function (newProps) {
+      this.setState({
+        toAnimate: this.state.toAnimate.concat(newProps.children)
```

Comment:
We might need to change this to `toAnimate: [].concat(newProps.children)`.  Reason being: right now we are continuously building off of the `toAnimate` state every time we receive new props.  Thus if we have a list of rooms and then we add to it we will end up with `toAnimate` having two lists of rooms in it.